hackathon_site/event/serializers.py

```diff
@@ -85,6 +85,26 @@ def update(self, instance: Profile, validated_data):
         # ProfileSerializer update function and we want to keep the logic between these two update functions separate.
         return serializers.ModelSerializer.update(self, instance, validated_data)
 
+    def create(self, validated_data):
+        try:
+            self.context["request"].user.profile
+            raise serializers.ValidationError("User already has profile")
+        except:
+
+            response_data = {"attended": True}
+            response_data["id_provided"] = False
+            response_data["acknowledge_rules"] = validated_data.pop(
+                "acknowledge_rules", False
+            )
+            response_data["e_signature"] = validated_data.pop("e_signature", "NULL")
+            response_data["user"] = self.context["request"].user
+            response_data["id"] = self.context["request"].user.id
```

Comment:
the id field is a primary key so its automatically generated by django when the profile is created, you don't need to set it here.